resources/views/auth/login.blade.php

```diff
@@ -8,103 +8,127 @@
-			$img.on('load', function(e){
-				loadPage(true);
-			});
-			$img.on('error', function() {
-				loadPage(false);
-			});
-			$img.css("display", "none").appendTo(document.body);
+	$(function() {
+		// Test content blocking on firefox:
+		let $img = $('<img/>').attr("src", "https://apps.facebook.com/favicon.ico");
+		$img.on('load', function(e){
+			loadPage(true);
 		});
-		function loadPage(isContentBlockingEnabled) {
-			if(!isContentBlockingEnabled) {
-				let msg = 'Please disable content blocking on this page to be able to sign in.';
-				console.error(msg);
-				$('#my-signin2').html(msg);
-				return false;
-			}
-			// let auth2;
-			gapi.load('auth2', function() {
-				var auth2 = gapi.auth2.init({
-					hosted_domain: 'itesm.mx',
-					prompt: 'select_account'
-				});
-				auth2.then(function() {
-					const hasUser = <?= Auth::user() === NULL ? 'false' : 'true' ?>;
-					if(auth2.isSignedIn.get() && !hasUser) {
-						auth2.signOut().then(function() {
-							loadBtn();
-						});
-					}
-					else {
-						setTimeout(loadBtn, 1200);
-					}
-				});
-			});
+		$img.on('error', function() {
+			loadPage(false);
+		});
+		$img.css("display", "none").appendTo(document.body);
+	});
+
+	function loadPage(isContentBlockingEnabled) {
+		if(!isContentBlockingEnabled) {
+			let msg = 'Please disable content blocking on this page to be able to sign in.';
+			console.error(msg);
+			$('#my-signin2').html(msg);
+			return false;
 		}
-		function loadBtn() {
-			gapi.signin2.render('my-signin2', {
-				'width': 240,
-				'height': 50,
-				'longtitle': true,
-				'theme': 'light',
-				'onsuccess': onSignIn,
-				'onfailure': onFailure
+		// let auth2;
+		gapi.load('auth2', function() {
+			var auth2 = gapi.auth2.init({
+				hosted_domain: 'itesm.mx',
+				prompt: 'select_account'
+			});
+			auth2.then(function() {
+				const hasUser = <?= Auth::user() === NULL ? 'false' : 'true' ?>;
+				if(auth2.isSignedIn.get() && !hasUser) {
+					auth2.signOut().then(function() {
+						loadBtn();
+					});
+				}
+				else {
+					setTimeout(loadBtn, 1200);
+				}
 			});
+		});
+	}
+
+	function loadBtn() {
+		gapi.signin2.render('my-signin2', {
+			'width': 240,
+			'height': 50,
+			'longtitle': true,
+			'theme': 'light',
+			'onsuccess': onSignIn,
+			'onfailure': onFailure
+		});
+	}
+
+	function showError(errorMessage) {
+		$('#errorReason').html(errorMessage);
+		$('#signInError').removeClass("hidden");
+	}
+
+	function onFailure(e) {
+		// Handle the error where user does not match domain
+		if(e.error) {
+			showError(errorMessageItesmAccount);
 		}
-		function onFailure(e) {
-			// Handle the error where user does not match domain
-			if(e.error) {
-				console.error(e);
-			}
-			if(e.type && e.type === "tokenFailed") {
-				alert('Could not sign you in using this account. Please use your \@itesm.mx account');
-			}
+		if(e.type && e.type === "tokenFailed") {
+			showError(errorMessageGeneric);
```

Comment:
Estos mensajes de error son al revés.

Reply:
Ah sí. Eso era lo que te preguntaba el otro día. El comment de not matched domain (@itesm) estaba en la parte de arriba pero el error es el de abajo. Entonces muevo el comment para abajo? 

```php
function onFailure(e) {
		// Handle the error where user does not match domain
		if(e.error) {
			showError(errorMessageItesmAccount);
		}
		if(e.type && e.type === "tokenFailed") {
			showError(errorMessageGeneric);
		}
	}
```

Reply:
Si please